tests/devices/test_default_mixed.py

```diff
@@ -448,9 +448,9 @@ def test_diag_init(self, x, tol):
         dev = qml.device("default.mixed", wires=nr_wires)
         kraus = dev._get_kraus(op)
         if op == CZ:
-            dev._apply_channel(kraus, wires=Wires([0, 1]))
+            dev._apply_diagonal_unitary(kraus, wires=Wires([0, 1]))
```

Comment:
This was an interesting bug revealed by the change: before, `_get_kraus` would not recognise that `CZ` is diagonal because `op` was a class, not an instance (and `isintance(op, DiagonalOperation)` returned False). So `_apply_channel` would work, even though for diagonal unitaries, `get_kraus` should return a 1-d array of eigenvalues that can only be processed by `_apply_diagonal_unitary`. 

A good sign that the change works!

Reply:
Good catch!

Reply:
nice!